rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/internal/LRARecoveryModule.java

```diff
@@ -177,7 +152,10 @@ private void processTransactionsStatus() {
                         doRecoverTransaction(currentUid);
                     }
                 } catch (ObjectStoreException e) {
-                    if (LRALogger.logger.isInfoEnabled()) {
+                    if (LRALogger.logger.isTraceEnabled()) {
+                        LRALogger.logger.tracef("failed to access transaction store %s: %s",
```

Comment:
This won't print the stacktrace as it uses the `tracef` definition of `tracef(String format, Object... params)` but it needs to be used the `tracef(Throwable t, String format, Object... params)`
Then printing "manually" the exception message with `e.getMessage()` is not necessary as the message will be automatically printed by logger when exception `e` is passed as `Throwable`.

It could be like

```
LRALogger.logger.tracef(e, "failed to access transaction store %s", currentUid);
```

And maybe one more question on the message. Is the `currentUid` meaning "an access of the transaction store". Should not be the message informs about not able to accessing some LRA UID or similar?

Reply:
1. The ObjectStore API does not say under what circumstances the exception is thrown since it depends upon which underlying store is in use and the reasons for a failure are varied and we certainly don't want to start adding logging logic for the different store types. Hence the generic message.

2. The log messages are for the benefit of the support organisation to figure out where the issue lies. The actual message text (excluding the values of variables) is less relevant (provided an engineer can identify where in the code the message was generated from).

3. Including the exception message in the trace message means that users who aren't interested in the call stack can still read the log messages and see what the precise reason for the error was (without having to scan down to the stacktrace).

But I did fix the log message text.

Reply:
Just for the technical merit I would mention that user don't need to scan down to the stacktrace to see the error message as it's always logged on the first line. For more details he needs to really go down to cause clauses but he has to go through exception stacktrace always for it.
Not a thing which I consider important.